bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/SagerWeatherCaster.java

```diff
@@ -0,0 +1,400 @@
+    }
+
+    private String getCompass() {
+        double step = 360.0 / NTZDIRECTIONS.length;
+        double b = Math.floor((this.currentBearing + (step / 2.0)) / step);
+        return NTZDIRECTIONS[(int) (b % NTZDIRECTIONS.length)];
+    }
+
+    private void updatePrediction() {
+        int zWind = Arrays.asList(usedDirections).indexOf(getCompass());
+        String d1 = "-";
+
+        if (zWind == 0) {
+            if (windEvolution == 3) {
+                d1 = "A";
+            } else if (windEvolution == 1) {
+                d1 = "B";
+            } else if (windEvolution == 2) {
+                d1 = "C";
+            }
+        } else if (zWind == 1) {
+            if (windEvolution == 3) {
+                d1 = "D";
+            } else if (windEvolution == 1) {
+                d1 = "E";
+            } else if (windEvolution == 2) {
+                d1 = "F";
+            }
+        } else if (zWind == 2) {
+            if (windEvolution == 3) {
+                d1 = "G";
+            } else if (windEvolution == 1) {
+                d1 = "H";
+            } else if (windEvolution == 2) {
+                d1 = "J";
+            }
+        } else if (zWind == 3) {
+            if (windEvolution == 3) {
+                d1 = "K";
+            } else if (windEvolution == 1) {
+                d1 = "L";
+            } else if (windEvolution == 2) {
+                d1 = "M";
+            }
+        } else if (zWind == 4) {
+            if (windEvolution == 3) {
+                d1 = "N";
+            } else if (windEvolution == 1) {
+                d1 = "O";
+            } else if (windEvolution == 2) {
+                d1 = "P";
+            }
+        } else if (zWind == 5) {
+            if (windEvolution == 3) {
+                d1 = "Q";
+            } else if (windEvolution == 1) {
+                d1 = "R";
+            } else if (windEvolution == 2) {
+                d1 = "S";
+            }
+        } else if (zWind == 6) {
+            if (windEvolution == 3) {
+                d1 = "T";
+            } else if (windEvolution == 1) {
+                d1 = "U";
+            } else if (windEvolution == 2) {
+                d1 = "V";
+            }
+        } else if (zWind == 7) {
+            if (windEvolution == 3) {
+                d1 = "W";
+            } else if (windEvolution == 1) {
+                d1 = "X";
+            } else if (windEvolution == 2) {
+                d1 = "Y";
+            }
+        } else if (currentBeaufort == 0) {
+            d1 = "Z";
+        }
+
+        String forecast = sagerForecaster
+                .get(d1 + String.valueOf(sagerPressure) + String.valueOf(pressureEvolution) + String.valueOf(nubes));
+        if (forecast != null) {
+            prevision = new Prevision(forecast);
+        } else {
+            prevision = null;
+        }
+    }
+
+    public String getForecast() {
+        if (prevision != null) {
+            char forecast = prevision.zForecast;
```

Comment:
You are going to receive warnings regarding prevision being nullable here and there are a couple of ways you can solve them.
One way is to use a local variable:
```suggestion
        Prevision prevision = this.prevision;
        if (prevision != null) {
            char forecast = prevision.zForecast;
```
The other way is to change the prevision field into Optional<Prevision>:
```suggestion
        if (prevision.isPresent()) {
            char forecast = prevision.get().zForecast;
```